pulsar-broker-common/src/main/java/org/apache/pulsar/broker/resources/LocalPoliciesResources.java

```diff
@@ -66,7 +66,7 @@ public CompletableFuture<Optional<CacheGetResult<LocalPolicies>>> getLocalPolici
     public CompletableFuture<Void> setLocalPoliciesWithVersion(NamespaceName ns, LocalPolicies policies,
                                                                Optional<Long> version) {
         try {
-            byte[] content = ObjectMapperFactory.getThreadLocal().writeValueAsBytes(policies);
+            byte[] content = ObjectMapperFactory.getMapper().writer().writeValueAsBytes(policies);
```

Comment:
How about having `ObjectMapperFactory.getWriter()` and `ObjectMapperFactory.getReader()` methods ?

Reply:
I was thinking of that, but the challenge is that there's also the Yaml versions and the ObjectMapper configured to filter null fields. That would result in a lot of similar methods. It's a tradeoff. 
Leaving ObjectMapperFactory as-is and creating a completely new class would be another option. I'd suggest that it's left as another PR to continue refactoring.